src/System.Linq.Expressions/src/System/Linq/Expressions/Compiler/ILGen.cs

```diff
@@ -875,25 +875,26 @@ private static void EmitNumericConversion(this ILGenerator il, Type typeFrom, Ty
             }
         }
 
-        private static void EmitNullableToNullableConversion(this ILGenerator il, Type typeFrom, Type typeTo, bool isChecked)
+        private static void EmitNullableToNullableConversion(this ILGenerator il, Type typeFrom, Type typeTo, bool isChecked, ILocalCache locals)
         {
             Debug.Assert(typeFrom.IsNullableType());
             Debug.Assert(typeTo.IsNullableType());
             Label labIfNull;
             Label labEnd;
-            LocalBuilder locFrom = il.DeclareLocal(typeFrom);
+            LocalBuilder locFrom = locals.GetLocal(typeFrom);
             il.Emit(OpCodes.Stloc, locFrom);
-            LocalBuilder locTo = il.DeclareLocal(typeTo);
+            LocalBuilder locTo = locals.GetLocal(typeTo);
             // test for null
             il.Emit(OpCodes.Ldloca, locFrom);
             il.EmitHasValue(typeFrom);
             labIfNull = il.DefineLabel();
             il.Emit(OpCodes.Brfalse_S, labIfNull);
             il.Emit(OpCodes.Ldloca, locFrom);
+            locals.FreeLocal(locFrom);
```

Comment:
I'd move release past `EmitGetValueOrDefault`. 
There is no difference, since `EmitGetValueOrDefault` does not require locals, but it makes the reader wonder why releasing before the use is safe.